### PR TITLE
Added configuration files and developed Maven sub-modules

### PR DESCRIPTION
**Pull request by NassimBtk**
Added configuration files and developed Maven sub-modules

Developed Maven modules:
* my-menu-tsps-core
* my-menu-tsps-web
* my-menu-tsps-lib
* my-menu-tsps-rest-api
* my-menu-tsps-plugin

Created docs folder

Created .github folder for workflow CI.

Updated .gitignore file

Update README.md file

### CODE REVIEW

File: my-menu-tsps-core/src/main/java/com/bmc/truesight/platform/etl/component/management/types/MyMenuComponent.java [hunk 1]
```diff
@@ -0,0 +1,323 @@
...
+
+	public User getMyMenuUser() {
+		return myMenuUser;
+	}
+
+	/**
+	 * @return The username configured in this Component, or "admin" if null or
+	 *         empty
+	 */
+	private String getUsername() {
```
Comment (bertysentry) [inline my-menu-tsps-core/src/main/java/com/bmc/truesight/platform/etl/component/management/types/MyMenuComponent.java]: I guess this method is not required. I haven't seen a Username attribute in the Component. However, we could show how to add such attribute. But not "Username". Maybe just something like "myAttribute".

File: my-menu-tsps-core/src/main/java/com/bmc/truesight/platform/etl/component/management/types/MyMenuComponent.java [hunk 2]
```diff
@@ -0,0 +1,323 @@
...
+		// There must be only one instance of the My Menu component
+		final List<IComponentType> myMenuComponentList = ComponentsDAO.getComponentsByType(COMPONENT_TYPE)
+				.orElse(new ArrayList<IComponentType>());
+		if (myMenuComponentList.indexOf(this) != 0) {
+			throw new ComponentManagementException("There must be only 1 instance of the My Menu Component.");
+		}
+
+		// Retrieve the User configured in the My Menu Component
+		try {
+			myMenuUser = new SecurityServiceImpl().getUser(super.getServerTenant(), getUsername());
```
Comment (bertysentry) [inline my-menu-tsps-core/src/main/java/com/bmc/truesight/platform/etl/component/management/types/MyMenuComponent.java]: I don't think it's necessary, as it's really for authentication matters, which was useful only in Monitoring Studio's case, where we do execute stuff on PATROL Agent "as a user"

File: my-menu-tsps-core/src/test/resources/my-menu-build.properties [hunk 3]
```diff
@@ -0,0 +1 @@
+version=${project.version}
```
Comment (bertysentry) [inline my-menu-tsps-core/src/test/resources/my-menu-build.properties]: This file is not required

File: my-menu-tsps-lib/src/main/java/com/bmc/sas/ngp/service/security/SecurityServiceImpl.java [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+ * as their "real" counterparts will be provided at runtime in the classpath of
+ * the JVM executing TSPS and our code.
+ *
+ * DO NOT CHANGE.
+ */
+package com.bmc.sas.ngp.service.security;
+
+import com.bmc.sas.domain.user.User;
+
+public class SecurityServiceImpl {
```
Comment (bertysentry) [inline my-menu-tsps-lib/src/main/java/com/bmc/sas/ngp/service/security/SecurityServiceImpl.java]: Maybe not necessary if you remove the corresponding call

File: my-menu-tsps-lib/src/main/java/com/bmc/truesight/api/dto/Component.java [hunk 5]
```diff
@@ -0,0 +1,171 @@
...
+ * as their "real" counterparts will be provided at runtime in the classpath of
+ * the JVM executing TSPS and our code.
+ *
+ * DO NOT CHANGE.
+ */
+package com.bmc.truesight.api.dto;
+
+import java.util.Set;
+
+public class Component {
```
Comment (bertysentry) [inline my-menu-tsps-lib/src/main/java/com/bmc/truesight/api/dto/Component.java:15]: OMG!! I didn't realize you would have to do this! I'm so sorry! And also happy that you manage to do it anyway :-)

File: my-menu-tsps-plugin/src/main/resources/conf/registry.xml [hunk 6]
```diff
@@ -0,0 +1,11 @@
+<?xml version="1.0" encoding="utf-8"?>
+<registry>
+	<resources>
+		<resource startup="false">
+			<className>com.mycompany.mymenu.tsps.controller.MyMenuDeviceProvider</className>
+			<targets>
+				<target authenticate="true" method="GET"><![CDATA[.*/my-menu/.*]]></target>
```
Comment (bertysentry) [inline my-menu-tsps-plugin/src/main/resources/conf/registry.xml:7]: I actually did not expect you to explain how to extend the REST API. But oh well, this is done, great! ;-)

File: my-menu-tsps-web/pom.properties [hunk 7]
```diff
@@ -0,0 +1,4 @@
+#Generated by Apache Maven ${maven.version}
+version=${project.version}
```
Comment (bertysentry) [inline my-menu-tsps-web/pom.properties]: Weird: this has been committed with Windows-style CR-LR end-of-lines (it's marked in red by GitHub `^M`). You should probably fix this.

File: my-menu-tsps-web/src/main/webapp/components/my-menu-devices-tsps/my-menu-devices-tsps.html [hunk 8]
```diff
@@ -0,0 +1,39 @@
+<!-- col-sm-12 is kinda mandatory here to avoid unnecessary scrollbars... -->
+<div class="my-menu-tsps col-sm-12">
```
Comment (bertysentry) [inline my-menu-tsps-web/src/main/webapp/components/my-menu-devices-tsps/my-menu-devices-tsps.html]: Please add a very simple `<bmc-title>` so future devs see how this works

File: my-menu-tsps-web/src/main/webapp/css/my-menu-tsps.css [hunk 9]
```diff
@@ -0,0 +1,34 @@
...
+  padding: 15px;
+  margin-top: 0.5em;
+}
+
+/* Adjust the size of the columns in the table */
+.my-menu-tsps .column-monitor {
+  width: 30%;
+  max-width: 300px;
+  min-width: 80px;
+  font-family: Verdana, Arial, sans-serif;
```
Comment (bertysentry) [inline my-menu-tsps-web/src/main/webapp/css/my-menu-tsps.css]: Do you really have to set *font-family*?

File: my-menu-tsps-web/src/main/webapp/css/my-menu-tsps.css [hunk 10]
```diff
@@ -0,0 +1,34 @@
...
+}
+
+#my-menu-dashboard {
+  color: rgb(178, 178, 178);
+  padding-left: 10px;
+  padding-top: 5px;
+  font-size: 3em;
+  font-family: OpenSans;
```
Comment (bertysentry) [inline my-menu-tsps-web/src/main/webapp/css/my-menu-tsps.css]: Same here. I'd prefer to encourage future devs to use existing classes and stuff :-)

File: my-menu-tsps-web/src/main/webapp/js/tsps-route.js [hunk 11]
```diff
@@ -0,0 +1,25 @@
...
+        permissions: "allow_uadmin_ui_login"
+      });
+    }
+  ]);
+  /**
+   * AngularJS constant: myMenuPermissions
+   * By default, myMenuPermissions say we're allowed to access stuff
+   * We ask our REST API what we're allowed to do
+   **/
+  angular
```
Comment (bertysentry) [inline my-menu-tsps-web/src/main/webapp/js/tsps-route.js]: This is not needed in this example, please remove.

**Review by bertysentry — CHANGES_REQUESTED** [review]
Brilliant work, @NassimBtk !
A few minor changes required. 
You will also need to document the installation part, clearly explaining that they need to "Add the Component" to their TSPS (not just install).
Thank you!

File: my-menu-tsps-lib/src/main/java/com/bmc/truesight/api/dto/Component.java (hunk 5 above)
Comment (NassimBtk) [reply my-menu-tsps-lib/src/main/java/com/bmc/truesight/api/dto/Component.java:15]: No problem @bertysentry  ;)

File: my-menu-tsps-web/src/main/webapp/css/my-menu-tsps.css (hunk 9 above)
Comment (NassimBtk) [reply my-menu-tsps-web/src/main/webapp/css/my-menu-tsps.css]: It is the same :D, but just to show that we can customize things ;)

Comment (NassimBtk) [reply my-menu-tsps-web/src/main/webapp/css/my-menu-tsps.css]: Removed

File: my-menu-tsps-web/src/main/webapp/components/my-menu-devices-tsps/my-menu-devices-tsps.html (hunk 8 above)
Comment (NassimBtk) [reply my-menu-tsps-web/src/main/webapp/components/my-menu-devices-tsps/my-menu-devices-tsps.html]: Ok let's use the BMC components 
```
<title-section> <title-bar> <title-display>
```

File: my-menu-tsps-lib/src/main/java/com/bmc/truesight/api/dto/ComponentAttributeEntity.java [hunk 12]
```diff
@@ -39,4 +103,146 @@ public String getAttributeName() {
 	public void setAttributeName(final String attributeName) {
 		this.attributeName = attributeName;
 	}
+
+	public String getAttributeValue() {
```
Comment (bertysentry) [inline my-menu-tsps-lib/src/main/java/com/bmc/truesight/api/dto/ComponentAttributeEntity.java:107]: Urgl... sorry again for the pain! ;-)

**Review by bertysentry — APPROVED** [review]

**Pull request merged**